### PR TITLE
feat(aws) add region for GuardDuty checks

### PR DESCRIPTION
**Pull request by kagahd**
### Context

Until this PR, the prowler reports of GuardDuty checks did not contain the region of the GuardDuty detector. This made it quite difficult to find the right place to verify and mitigate the reported risks.

### Description

This PR adds to the prowler report the region of the analyzed GuardDuty detector.

### Checklist

- Are there new checks included in this PR? **No**
- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [x] Review if backport is needed.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jfagoagas** [comment]
Hello @kagahd, I'm just curious about why the `finding.region` is not enough for you to get the GuardDuty detector region. Even with that I'm fine with adding the region in the `result_extended` but I think it's a little redundant.

File: prowler/providers/aws/services/guardduty/guardduty_no_high_severity_findings/guardduty_no_high_severity_findings.py [hunk 1]
```diff
@@ -13,10 +13,10 @@ def execute(self):
                 report.resource_arn = detector.arn
                 report.resource_tags = detector.tags
                 report.status = "PASS"
-                report.status_extended = f"GuardDuty detector {detector.id} does not have high severity findings."
```
Comment (jfagoagas) [inline prowler/providers/aws/services/guardduty/guardduty_no_high_severity_findings/guardduty_no_high_severity_findings.py:16]: Probably this check will need to modify their tests.

**Comment by kagahd** [comment]
> Hello @kagahd, I'm just curious about why the `finding.region` is not enough for you to get the GuardDuty detector region.

Hi @jfagoagas, you're completely right! I oversaw that the region is already in the report. So there is no need to add it again to `report.status_extended`. 
Sorry for the PR. It can be closed unmerged.

**Comment by jfagoagas** [comment]
> > Hello @kagahd, I'm just curious about why the `finding.region` is not enough for you to get the GuardDuty detector region.
> 
> Hi @jfagoagas, you're completely right! I oversaw that the region is already in the report. So there is no need to add it again to `report.status_extended`. Sorry for the PR. It can be closed unmerged.

It's totally fine, thank you for using and taking care of Prowler ❤️

**Pull request closed without merging**